ecs-agent/csiclient/csi_client.go

```diff
@@ -0,0 +1,78 @@
+package csiclient
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	PROTOCOL = "unix"
+)
+
+type csiClient struct {
+	csiSocket string
+}
+
+func NewCSIClient(socketIn string) csiClient {
+	return csiClient{csiSocket: socketIn}
+}
+
+// Used for testing and integration
+// TODO update with stats packing
+func (cc *csiClient) GetVolumeMetrics(volumeId string, hostMountPath string) (int64, error) {
```

Comment:
this is still for local testing only? We should return totalBytes and usedBytes ideally.

Reply:
And this pkg `csiclient` will hold all shared interfaces used to communicate with the csi driver daemon, right?

Reply:
(yes this will be updated as we integrate with telemetry -- I'll consider this a casualty of not having a feature branch for this work.)

Reply:
this package will hold specifically CSI "Node" client calls, yes.